packages/expo-router/src/modal/Modal.tsx

```diff
@@ -135,10 +144,11 @@ export function Modal(props: ModalProps) {
   useEffect(() => {
     if (currentModalId && visible) {
       updateModal(currentModalId, {
+        detentIndex,
         component: children,
       });
     }
-  }, [children]);
+  }, [children, detentIndex]);
```

Comment:
I would remove the `index` from here. It is meant to be initial, so it should not be changed while the modal is open